src/unixTerminal.ts

```diff
@@ -231,9 +232,18 @@ export class UnixTerminal extends Terminal {
   }
 
   public kill(signal?: string): void {
-    try {
-      process.kill(this.pid, signal || 'SIGHUP');
-    } catch (e) { /* swallow */ }
+    signal = signal || 'SIGHUP';
+    if (signal in os.constants.signals) {
+      try {
+        if (pty.kill && signal !== 'SIGHUP') {
```

Comment:
Probably don't need the `pty.kill` check, also a comment explaining why this is needed would be good here.